crates/precompile/src/utilities.rs

```diff
@@ -1,42 +1,148 @@
-use alloc::vec::Vec;
+use alloc::borrow::Cow;
 use core::cmp::min;
 
-/// Get an array from the data, if data does not contain `start` to `len` bytes,
-/// add right padding with zeroes.
+/// Right-pads the given slice at `offset` with zeroes until `LEN`.
+///
+/// Returns the first `LEN` bytes if it does not need padding.
 #[inline(always)]
-pub fn get_right_padded<const LEN: usize>(data: &[u8], offset: usize) -> [u8; LEN] {
-    let mut padded = [0; LEN];
-    let start = min(offset, data.len());
-    let end = min(start.saturating_add(LEN), data.len());
-    padded[..end - start].copy_from_slice(&data[start..end]);
-    padded
+pub fn get_right_padded<const LEN: usize>(data: &[u8], offset: usize) -> Cow<'_, [u8; LEN]> {
```

Comment:
As we have `right_pad`, would rename this to be more aligned, something like `right_par_with_offset`